src/resources/assets/js/ol/MagicSamInteraction.js

```diff
@@ -209,6 +239,7 @@ class MagicSamInteraction extends PointerInteraction {
 
         let contour = MagicWand.traceContours(imageData)
             .filter(c => !c.inner)
+            .filter(c => isPointInsideContour(c, pointCoords[0], pointCoords[1]))
```

Comment:
I was going to say that the OpenLayers Polygon geometry has a `intersectsCoordinate()` method that could be used for this. But the coordinates are no Polygon at this point. However, I traced the OpenLayers logic to [this function](https://github.com/openlayers/openlayers/blob/v5.3.0/src/ol/geom/flat/contains.js#L37). Is this the same as yours? Then we could just import it here.

Reply:
If you flatten the contours before then it should be the same.